packages/compass-workspaces/src/components/workspaces.tsx

```diff
@@ -105,159 +102,117 @@ const CompassWorkspaces: React.FunctionComponent<CompassWorkspacesProps> = ({
-          const info = collectionInfo[namespaceId] ?? {};
-          const { isTimeSeries, isReadonly, sourceName, isNonExistent } = info;
-          const connectionName =
-            getConnectionById(tab.connectionId)?.title || '';
-          const collectionType = isTimeSeries
-            ? 'timeseries'
-            : isReadonly
-            ? 'view'
-            : 'collection';
-          // Similar to what we have in the collection breadcrumbs.
-          const tooltip: Tooltip = [
-            ['Connection', connectionName || ''],
-            ['Database', database],
-          ];
-          if (sourceName) {
-            tooltip.push(['View', collection]);
-            tooltip.push(['Derived from', toNS(sourceName).collection]);
-          } else if (tab.editViewName) {
-            tooltip.push(['View', toNS(tab.editViewName).collection]);
-            tooltip.push(['Derived from', collection]);
-          } else {
-            tooltip.push(['Collection', collection]);
-          }
-          return {
-            id: tab.id,
-            connectionName,
-            type: tab.type,
-            title: collection,
-            tooltip,
-            iconGlyph:
-              collectionType === 'view'
-                ? 'Visibility'
-                : collectionType === 'timeseries'
-                ? 'TimeSeries'
-                : isNonExistent
-                ? 'EmptyFolder'
-                : 'Folder',
-            'data-namespace': ns,
-            tabTheme: getThemeOf(tab.connectionId),
-            ...(isNonExistent && {
-              className: nonExistantStyles,
-            }),
-          } as const;
+          const { isNonExistent: collectionDoesNotExist } =
+            collectionInfo[namespaceId] ?? {};
+          isNonExistent = collectionDoesNotExist;
         }
       }
-    });
-  }, [tabs, collectionInfo, databaseInfo, getThemeOf, getConnectionById]);
 
-  const activeTabIndex = tabs.findIndex((tab) => tab === activeTab);
-  const WorkspaceComponent = getWorkspacePluginByName(activeTab?.type);
+      const PluginTabComponent = headerFn({
+        ...(isNonExistent !== undefined ? { isNonExistent } : {}),
+        ...tab,
+      });
 
-  const onCreateNewTab = useCallback(() => {
-    onCreateTab(openOnEmptyWorkspace);
-  }, [onCreateTab, openOnEmptyWorkspace]);
+      return {
+        id: tab.id,
+        renderTab: (workspaceTabProps: WorkspaceTabCoreProps) => (
+          <ErrorBoundary
+            displayName={tab.type}
+            onError={(error, errorInfo) => {
+              log.error(
+                mongoLogId(1_001_000_360),
+                'Workspace',
+                'Rendering workspace tab header failed',
+                { name: tab.type, error: error.message, errorInfo }
+              );
+            }}
+          >
+            <WorkspaceTabContextProvider
+              tab={tab}
+              sectionType="tab-title"
+              onNamespaceNotFound={onNamespaceNotFound}
```

Comment:
Should be passing this one only once, it doesn't make that much difference, but otherwise this gets called twice. We can probably adjust the types of WorkspaceTabContextProvider to make this prop required only if sectionType is content or something like that